ddtrace/settings/_core.py

```diff
@@ -13,6 +17,22 @@
 FLEET_CONFIG, LOCAL_CONFIG = get_configuration_from_disk()
 
 
+class DDConfig(Env):
+    """Provides support for loading configurations from multiple sources."""
+
+    def __init__(
+        self,
+        source: Optional[Dict[str, str]] = None,
+        parent: Optional["Env"] = None,
+        dynamic: Optional[Dict[str, str]] = None,
+    ) -> None:
+        self.fleet_source = FLEET_CONFIG
+        self.local_source = LOCAL_CONFIG
+        # Order of precedence: local stable config < environment variables < fleet stable config < provided source
+        source = ChainMap(source or {}, self.fleet_source, os.environ, self.local_source)  # type: ignore
+        super().__init__(source=source, parent=parent, dynamic=dynamic)
```

Comment:
```suggestion
        full_source = ChainMap(source or {}, self.fleet_source, os.environ, self.local_source)  # type: ignore
        super().__init__(source=full_source, parent=parent, dynamic=dynamic)
```
If `source` is `os.environ` the resolution order would change 🤔

Reply:
I moved it to the end. fleet automation configs should always take precedence. We shouldn't allow folks to override this source